packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -132,12 +135,28 @@ export const useCheckoutPage = (props = {}) => {
         setReviewOrderButtonClicked(false);
     }, [setReviewOrderButtonClicked]);
 
+    const scrollShippingInformationIntoView = useCallback(() => {
+        if (shippingInformationRef.current) {
+            shippingInformationRef.current.scrollIntoView({
+                behavior: 'smooth'
+            });
+        }
+    }, [shippingInformationRef]);
```

Comment:
This callback is basically duplicated and will be for each step that gets this new treatment. Don't spend too much time optimizing, but I wonder if there is a way to make this more generic. We already do have a `done` callback but it is used in a slightly different way (refresh of page vs "submit" of mutation). Hmmm...

Reply:
Yeah, the problem with the done callbacks is, they get called whenever the section is done, which includes page refresh as well. So I couldn't use them.